L1Trigger/L1THGCal/interface/hw_interfaces/HGCalHardwareInterface.h

```diff
@@ -13,58 +13,46 @@ class HGCalHardwareInterface{
   struct ClusterDataFormat{
     enum CDF {ET = 0, CenterY = 8, CenterX = 20, Flags = 32, SizeY = 38, SizeX = 46, LocalMax = 54, NoTriggerCells = 56, LocalMax1RelX = 64,  LocalMax0ET = 72, LocalMax0RelY = 80, LocalMax0RelX = 88, LocalMax2RelY = 96,  LocalMax2RelX = 104,  LocalMax1ET = 112,  LocalMax1RelY = 120,  LocalMax3ET = 128, LocalMax3RelY = 136,  LocalMax3RelX = 144,  LocalMax2ET = 152 };
   };
-  struct MultiClusterDataFormat{
+  struct MulticlusterDataFormat{
     enum MCDF {BHEFraction = 0, EEEFraction = 8, ET = 16, Z = 32, Phi = 42, Eta = 53, Flags = 64, QFlags = 76, NoTriggerCells = 88, TBA = 96, MaxEnergyLayer = 122};
   };
   
   typedef std::vector<uint32_t> ClusterDataContainer;
   typedef std::map<int,int>     DataFormatMap;
 
-  HGCalHardwareInterface(l1t::HGCalCluster * HGCC){
-    this->HGCC_  = HGCC;
-  };
-
-  HGCalHardwareInterface(l1t::HGCalMulticluster * HGCMC){
-    this->HGCMC_ = HGCMC;
-
-  };
-
-  bool DecodeMulticlusterData (const ClusterDataContainer& DataToBeDecoded);
+  bool decodeMulticlusterData (const ClusterDataContainer& DataToBeDecoded, l1t::HGCalMulticluster & HGCMC) const;
 	
-  bool EncodeMulticlusterData (ClusterDataContainer& DataToBeEncoded);
+  bool encodeMulticlusterData (ClusterDataContainer& DataToBeEncoded, const l1t::HGCalMulticluster & HGCMC) const;
 
-  bool DecodeClusterData (const ClusterDataContainer& DataToBeDecoded);
+  bool decodeClusterData (const ClusterDataContainer& DataToBeDecoded, l1t::HGCalCluster & HGCC) const;
 	
-  bool EncodeClusterData (ClusterDataContainer& DataToBeEncoded);
-
+  bool encodeClusterData (ClusterDataContainer& DataToBeEncoded, const l1t::HGCalCluster & HGCC) const;
   
  private:
         
   HGCalHardwareInterface(){};
   
-  l1t::HGCalMulticluster * HGCMC_;
-  l1t::HGCalCluster      * HGCC_;
 
-  /* Object required to have a low-latency data encode / decode scheme */
-  ClusterDataContainer MultiClusterData_;
+  /* Objects required to have a low-latency data encode / decode scheme */
+  static const DataFormatMap multiclusterDataMap_;
```

Comment:
The different maps are for 2D vs 3D clusters now, not clusters vs multiclusters, right?